rfc/0008-builtin-blocktypes/README.md

```diff
@@ -0,0 +1,212 @@
+<!--
+SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+
+SPDX-License-Identifier: AGPL-3.0-only
+-->
+
+# RFC 0008: Builtin blocktypes
+
+|             |                      |
+|-------------|----------------------|
+| Feature Tag | `builtin-blocktypes` |
+| Status      | `DISCUSSION`              | <!-- Possible values: DRAFT, DISCUSSION, ACCEPTED, REJECTED -->
+| Responsible | `@felix-oq`          |
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+
+Blocktypes that are built into the language are declared syntactically by denoting IO types and their properties.
+This change does not change the semantics, but it can serve as a foundation for composite block types in an upcoming RFC.
+
+## Motivation
+
+Currently, all blocktypes only exist implicitly in the language.
+To make their definition more explicit, we should support the declaration of builtin blocktypes (i.e. blocktypes that are built into the language).
+This allows users to look up IO types and properties without having to open the documentation pages.
+
+## Explanation
+
+- `builtin` keyword to indicate that the blocktype is built into the language
+  - Produces an error if the blocktype is unknown
+- `input` / `output` keywords to type the input and output of the blocktype
```

Comment:
I'd encourage inputs and outputs with an array with strings 

inputs: [
  "default" oftype file
]

That would allow us to be more future proof if (when) we move to multiple in/outputs but might be more complex right now. I am open to both, just leaving it as feedback for your consideration. Classic agile SE would teach us to not do future proofing work and just go with single in/outputs for now.

Reply:
I'd emphasize to avoid introducing more "artificial" collection properties to M2 concepts (like the `constraints` collection in valuetypes) that are not defined in the code.

I'd rather use a syntax similar to properties to define multiple inputs / outputs:
```
// default input:
input oftype sheet;

// additional named input:
input myInput oftype sheet;
```

And finally I want to highlight that modelling multiple inputs/outputs is considered beyond the scope of this RFC.

Reply:
I think it would be weird to have two collection type syntaxes. Even though I can see the difference here with the additional named input. But the same could have been said for constraints for example. I am happy with going with the single input oftype sheet for now in any case though, let's commit to that and not discuss multiple inputs here you are right.